plugins/basic/python/tk_maya_basic/plugin_logic.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (c) 2016 Shotgun Software Inc.
```

Comment:
All these files seem to be namespaced as `plugin_<name>.py`. They're all bootstrap related though. 

This file in particular seems like it might be combined with what's currently in `plugin_engine.py` as just `bootstrap.py`? Seems like the only client of this file is `shotgun.py`. 

Maybe these top-level methods could be exposed through `bootstrap.py` and then rename some of the components accordingly? My biggest concern is having multiple files in here with `shutdown()` and `bootstrap()` methods.

Maybe `plugin_logging.py` becomes `bootstrap_logging.py`?